metric-anomaly-task-manager/src/main/java/org/hypertrace/alert/engine/metric/anomaly/task/manager/job/DbRuleSource.java

```diff
@@ -0,0 +1,147 @@
+package org.hypertrace.alert.engine.metric.anomaly.task.manager.job;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.protobuf.Value;
+import com.google.protobuf.util.JsonFormat;
+import com.typesafe.config.Config;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+import org.hypertrace.alert.engine.metric.anomaly.datamodel.rule.source.RuleSource;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc.ConfigServiceBlockingStub;
+import org.hypertrace.config.service.v1.ContextSpecificConfig;
+import org.hypertrace.config.service.v1.GetAllConfigsRequest;
+import org.hypertrace.core.documentstore.Document;
+import org.hypertrace.core.documentstore.JSONDocument;
+import org.hypertrace.core.grpcutils.client.RequestContextClientCallCredsProviderFactory;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.core.serviceframework.spi.PlatformServiceLifecycle;
+import org.hypertrace.notification.config.service.v1.GetAllNotificationRulesRequest;
+import org.hypertrace.notification.config.service.v1.NotificationRule;
+import org.hypertrace.notification.config.service.v1.NotificationRuleConfigServiceGrpc;
+
+public class DbRuleSource implements RuleSource {
+
+  private static final String EVENT_CONDITION_DATA_KEY = "eventConditionData";
+  private static final String EVENT_CONDITION_MUTABLE_DATA_KEY = "eventConditionMutableData";
+  private static final String NOTIFICATION_RULE_MUTABLE_DATA_KEY = "notificationRuleMutableData";
+  private static final String METRIC_ANOMALY_DATA_KEY = "metricAnomalyEventCondition";
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  private final ConfigServiceBlockingStub configServiceBlockingStub;
+  private final NotificationRuleConfigServiceGrpc.NotificationRuleConfigServiceBlockingStub
+      notificationRuleStub;
+  private final String resourceNamespace;
+  private final String resourceName;
+  private final List<String> tenantIds;
+
+  public DbRuleSource(Config dataStoreConfig, PlatformServiceLifecycle lifecycle) {
+    ManagedChannel configChannel =
+        ManagedChannelBuilder.forAddress(
+                dataStoreConfig.getString("config.service.host"),
```

Comment:
nit: like others can we define a private static constant for all string - `config.service.host`, `resourceName` and all?